tests/lib/c_lib/src/main.c

```diff
@@ -791,6 +791,49 @@ void test_time(void)
 	zassert_not_null(gmtime_r(&tests4, &tp), "gmtime_r failed");
 }
 
+/**
+ *
+ * @brief test rand functions
+ *
+ */
+void test_rand(void)
+{
+	int a;
+
+	a = rand();
+	zassert_equal(a, 12345, "rand failed");
+}
+
+/**
+ *
+ * @brief test srand functions
+ *
+ */
+void test_srand(void)
```

Comment:
These tests look good, but ideally, the test would verify that rand() results after calling srand() with the same value are reproducible (per the definition of srand() semantics). I.e.: call srand() with one value, and check that a few (2-3) rand() calls give some values, then call srand() with another value, and call rand() a few times and check that values are different from 1st time. Then call srand() again with the initial value, and check that rand() give the same values as first time.

(That's in addition to the existing testcases - again, they're fine.)

Reply:
`zephyr/tests/lib/c_lib/README.txt` states:
> It is intended to catch issues in which a library is completely absent or non-functional, and is NOT intended to be a comprehensive test suite of all functionality provided by the libraries.

Regardless, I've added a reproducibility test for `rand()`.

Reply:
Sure, that was just a suggestion, as long as you went to write a more or less detailed test at all. I appreciate adding that test.